lib/token.h

```diff
@@ -49,6 +49,65 @@ struct TokensFrontBack {
     Token *back;
 };
 
+struct TokenImpl {
```

Comment:
Breaking out the members to a struct.. I fail to see how that makes the code faster. Could you explain?

The `deleteNext` seems to be faster now but was that a bottleneck?

Reply:
Making the Token class smaller means it's cheaper to pass around and more of them can fit in the CPU cache.  Only members that are not always used should be in the struct, because accessing them is slightly slower, but putting them in the struct means they don't have to be in the cache when they aren't needed.  This specific part of the change decreased running time by just under 20%.